### ISSUE TITLE
Service fails to start after upgrade to 20.07

### ISSUE THREAD

**Issue by mkl262** [processone/ejabberd#3457]
## Environment

- ejabberd version: 20.07
- Erlang version: `Earlang/OTP 21`
- OS: Debian Buster
- Installed from: official binary installer

I upgraded my ejabberd node according to the [procedure](https://docs.ejabberd.im/admin/upgrade/) from 19.09.1, and after restarting the service, it fails. I have attached the error and crash logs.
Also, even though that I see that the process tree is running in the server, it accepts connections, and writes to the log, ejabberdctl reports that the server is down.

[crash.log](https://github.com/processone/ejabberd/files/5734915/crash.log)
[error.log](https://github.com/processone/ejabberd/files/5734916/error.log)

Thank you.

**Comment by licaon-kter** [comment]
ejabberd.log says what?

**Comment by mkl262** [comment]
2020-12-23 11:19:01.595 [info] <0.64.0> SIGTERM received - shutting down
2020-12-23 11:19:01.595 [error] <0.872.0> gen_server memsup terminated with reason: {port_died,normal}
2020-12-23 11:19:01.595 [error] <0.872.0> CRASH REPORT Process memsup with 0 neighbours exited with reason: {port_died,normal} in gen_server:handle_common_reply/8 line 751
2020-12-23 11:19:01.595 [error] <0.871.0> Supervisor os_mon_sup had child memsup started with memsup:start_link() at <0.872.0> exit with reason {port_died,normal} in context child_terminated
2020-12-23 11:19:01.595 [info] <0.106.0>@ejabberd_listener:stop_listener:402 Stop accepting TCP connections at [::]:5222 for ejabberd_c2s
2020-12-23 11:19:01.596 [info] <0.106.0>@ejabberd_listener:stop_listener:402 Stop accepting TCP connections at [::]:5269 for ejabberd_s2s_in
2020-12-23 11:19:01.596 [info] <0.106.0>@ejabberd_listener:stop_listener:402 Stop accepting TCP connections at [::]:5443 for ejabberd_http
2020-12-23 11:19:01.596 [info] <0.106.0>@ejabberd_listener:stop_listener:402 Stop accepting TCP connections at [::]:5280 for ejabberd_http
2020-12-23 11:19:01.596 [info] <0.106.0>@ejabberd_listener:stop_listener:402 Stop accepting TCP connections at [::]:9334 for ejabberd_http

**Comment by mkl262** [comment]
It happens exactly 5 minutes after the start

**Comment by licaon-kter** [comment]
Received shutting down means you've stopped it.

Can you post the log of a start?0

**Comment by mkl262** [comment]
[ejabberd.log](https://github.com/processone/ejabberd/files/5735092/ejabberd.log)

**Comment by licaon-kter** [comment]
Also how do you start it?

**Comment by mkl262** [comment]
systemctl start ejabberd

**Comment by mkl262** [comment]
If i run `ejabberdctl start` it starts, but it crashes with the same error.
And if I try to run `ejabberdctl started` it gets stuck.

**Comment by licaon-kter** [comment]
So even if you don't do any API calls (like the logs show) it stops after 5 mins?

**Comment by mkl262** [comment]
Yes, with the same error

**Comment by Neustradamus** [comment]
@mkl262: What is your ejabberd.yml (Please remove password)

Have you tested with ejabberd 20.12 too?

**Comment by mkl262** [comment]
[ejabberd.yml.txt](https://github.com/processone/ejabberd/files/5738780/ejabberd.yml.txt)

**Comment by prefiks** [comment]
You are using system erlang right? Do you have erlang-os-mon installed on your system?

**Comment by mkl262** [comment]
No, I dont have system erlang, Im installing ejabberd from official binary installer.

**Comment by prefiks** [comment]
Could you check you syslog to see if there is any info about memsup or cpu_sup process in it?

Looks like for some reason erlang module that tracks system memory presure is crashing, I don't remember seeing that before maybe you system logs will show some error which will allow us to see why.
